pkg/argocd/update.go

```diff
@@ -210,8 +210,19 @@ func UpdateApplication(updateConf *UpdateConfiguration) ImageUpdaterResult {
 				result.NumErrors += 1
 				continue
 			} else {
-				imgCtx.Infof("Successfully updated image '%s' to '%s', but pending spec update (dry run=%v)", updateableImage.GetFullNameWithTag(), updateableImage.WithTag(latest).GetFullNameWithTag(), updateConf.DryRun)
+				message := fmt.Sprintf("Successfully updated image '%s' to '%s', but pending spec update (dry run=%v)", updateableImage.GetFullNameWithTag(), updateableImage.WithTag(latest).GetFullNameWithTag(), updateConf.DryRun)
+				imgCtx.Infof(message)
 				result.NumImagesUpdated += 1
+				if !updateConf.DryRun {
+					annotations := map[string]string{
+						"image-old": updateableImage.GetFullNameWithTag(),
+						"image-new": updateableImage.WithTag(latest).GetFullNameWithTag(),
+					}
+					_, err = updateConf.KubeClient.CreateApplicationEvent(&updateConf.UpdateApp.Application, "ImageUpdated", message, annotations)
+					if err != nil {
+						imgCtx.Warnf("Event could not be sent: %v", err)
+					}
+				}
```

Comment:
I think this should go further down at https://github.com/argoproj-labs/argocd-image-updater/blob/af590d33b782c339f4bdeb06a7fb79a313703e19/pkg/argocd/update.go#L245, after changes have been committed successfully to Argo CD and/or Git.

Also, because it is possible that we do not have a Kubernetes client, we should explicitly test for `updateConf.KubeClient != nil` before trying to use it.

Reply:
Hmmm.... you are right. But I'd need to store the list of updated images in order to use that information in the message. Correct?

Reply:
Yes, we'd probably need to store it in some kind of temporary data structure to access it later on.

Also, sorry, I forgot to mention that you'll need to run `make manifests` (requires `kustomize` installed) after changing the role in the kustomize source manifests and also push those changes to Git, because we'll provide a pre-rendered `install.yaml` and the CI checks whether it differs from the manifests generated during CI run.